src/guardianSigner.ts

```diff
@@ -0,0 +1,47 @@
+import { ISignable, ISignature } from "./interface";
+import { UserSecretKey } from "./userKeys";
+import { UserSigner } from "./userSigner";
+import { Signature } from "./signature";
+import { ErrSignerCannotSign } from "./errors";
+
+/**
+ * ed25519 signer
+ */
+export class GuardianSigner extends UserSigner {
+    private readonly guardianSecretKey: UserSecretKey;
+
+    constructor(guardianSecretKey: UserSecretKey) {
+        super(guardianSecretKey)
+        this.guardianSecretKey = guardianSecretKey
+    }
+
+/**
+ * Signs a message.
+ * @param signable the message to be signed (e.g. a {@link Transaction}).
+ */
+    async guard(signable: ISignable): Promise<void> {
+        try {
+            this.tryGuard(signable);
+        } catch (err: any) {
+            throw new ErrSignerCannotSign(err);
+        }
+    }
+
+    private tryGuard(signable: ISignable) {
+        let ownerSignature = signable.getSignature()
+        let bufferToSign = signable.serializeForSigning();
+        let guardianSignatureBuffer = this.guardianSecretKey.sign(bufferToSign);
+        let guardianSignature = new Signature(guardianSignatureBuffer);
+
+        this.addOwnerSignature(signable, ownerSignature)
+        this.doApplySignature(signable, guardianSignature);
+    }
+
+    protected doApplySignature(signable: ISignable, guardianSignature: ISignature): void {
```

Comment:
for some method you explicitly put the "void" return value, but for others this is not the case. Can we have remove it so we can keep the same code style on the whole project?

Reply:
I would add `: void` to all functions which don't return a value for easing debugging
:smile: